src/features/community/merchant/actions/loadFrogs.ts

```diff
@@ -0,0 +1,127 @@
+import { communityContracts } from "features/community/lib/communityContracts";
+import frog_unrevealed from "assets/nfts/frogs/frog_unrevealed.gif";
+
+type Request = {
+  frogId: number[];
+};
+
+export type Frog = {
+  pixel_image: string;
+  attributes?: any;
+  [key: string]: any;
+};
+
+export async function loadFrogs() {
+  let frogIds = await communityContracts.getFrog().getFrogIds();
+
+  // for testing
+  // if (false) {
+  if (await isFrogPreview()) {
+    if (frogIds.length > 6) {
+      frogIds = frogIds.slice(0, 6);
+    }
+
+    const result = frogIds.map(() => ({
+      pixel_image: frog_unrevealed,
+      name: "Unrevealed",
+    }));
+
+    return result;
+  } else {
+    const result = frogIds.map(
+      async (id) => await getFrogMetadata({ frogId: id })
+    );
+    const res: Frog[] = await Promise.all(result);
+
+    let filteredFrogs = res;
+
+    // only run filter if there is more than 1 frog in wallet
+    if (frogIds.length > 1) {
+      filteredFrogs = filterFrogs(res);
+    }
+
+    console.log("filtered frogs:", filteredFrogs);
+    return filteredFrogs;
+  }
+}
+
+export async function getFrogMetadata(request: Request) {
+  const headers = new Headers();
+
+  headers.append("Content-Type", "application/json");
+  headers.append("Accept", "application/json");
+
+  const frogBaseUri = await communityContracts.getFrog().getBaseUri();
```

Comment:
<img width="348" alt="Screen Shot 2022-08-29 at 8 48 26 am" src="https://user-images.githubusercontent.com/11745561/187097729-238f1f09-64ad-4d09-9804-423fdec8a4ad.png">

Calling the `tokenUri` directly on an ERC721 contract will return the full version of the URL. This means you don't need to do concatenation below.

Reply:
I forgot there's a tokenURI function 😅 thanks adam